lib/vanagon/platform/rpm/eos.rb

```diff
@@ -0,0 +1,43 @@
+class Vanagon
+  class Platform
+    class RPM
+      class EOS < Vanagon::Platform::RPM
+        # The specific bits used to generate an SWIX package for a given project
+        #
+        # @param project [Vanagon::Project] project to build a SWIX package of
+        # @return [Array] list of commands required to build the SWIX package
+        # for the given project from an rpm
+        def generate_package(project)
+          if project.platform.package_type == 'rpm'
+            super
+          else
+            target_dir = project.repo ? output_dir(project.repo) : output_dir
+
+            commands = super(project)
+            pkgname_swix = package_name(project)
+            pkgname_rpm = pkgname_swix.sub(/swix$/, 'rpm')
+            commands += ["echo 'format: 1' > ./output/#{target_dir}/manifest.txt",
+            "echo \"primaryRpm: #{pkgname_rpm}\" >> ./output/#{target_dir}/manifest.txt",
+            "echo #{pkgname_rpm}-sha1: `sha1sum ./output/#{target_dir}/#{pkgname_rpm}`",
+            "cd ./output/#{target_dir}/ && zip #{pkgname_swix} manifest.txt #{pkgname_rpm}",
+            "rm ./output/#{target_dir}/manifest.txt ./output/#{target_dir}/#{pkgname_rpm}"]
+
+            commands
+          end
+        end
+
+        # Method to derive the package name for the project
+        #
+        # @param project [Vanagon::Project] project to name
+        # @return [String] name of the SWIX package for this project
+        def package_name(project)
+          if project.platform.package_type == 'rpm'
+            super
+          else
+            "#{project.name}-#{project.version}-#{project.release}.#{os_name}#{os_version}.#{project.noarch ? 'noarch' : @architecture}.swix"
+          end
```

Comment:
Are there any other "if/else" decisions likely to be made for this sort of paradigm in the future? If so, this pattern portends a lot of methods with cascading "if/else" decisions in them; is it worth abstracting this out into a module/mix-in pattern now so that the groundwork is there in the future or are we OK saying "we will cross that bridge when we get there"?